media-source-respec.html

```diff
@@ -412,28 +418,30 @@ <h3>Attributes</h3>
               </p>
             </li>
             <li>If the {{MediaSource/readyState}} attribute is not in the {{ReadyState/""open""}} state then throw an {{InvalidStateError}} exception and abort these steps.</li>
-            <li>Create a new <a>SourceBuffer</a> object and associated resources.</li>
-            <li>Set the {{SourceBuffer/[[generate timestamps flag]]}} on the new object to the value in the "Generate
-              Timestamps Flag" column of the byte stream format registry [[MSE-REGISTRY]] entry that is associated with
+            <li>
+              Let |buffer| be a new instance created as follows:
+              <ul>
+                <li>
+                  If [=this=]'s {{MediaSource/[[is actively managed flag]]}} is false, a newly created {{SourceBuffer}} instance along with its associated resources.
+                </li>
+                <li>
+                  Otherwise, a newly created {{ManagedSourceBuffer}} instance and its associated resources.
+                </li>
+              </ul>
+            </li>
+            <li>
+              Set |buffer|'s {{SourceBuffer/[[generate timestamps flag]]}} to the value in the "Generate
+              Timestamps Flag" column of the [[[MSE-REGISTRY]]] entry that is associated with
               |type|.
-            </li><li>
-              <dl class="switch">
-                <dt>If the {{SourceBuffer/[[generate timestamps flag]]}} equals true:</dt>
-                <dd>
-                  Set the {{SourceBuffer/mode}} attribute on the new object to
-                  {{AppendMode/""sequence""}}.
-                </dd>
-                <dt>Otherwise:</dt>
-                <dd>
-                  Set the {{SourceBuffer/mode}} attribute on the new object to
-                  {{AppendMode/""segments""}}.
-                </dd>
-              </dl>
             </li>
-            <li>Add the new object to {{MediaSource/sourceBuffers}} and [=queue a task=] to [=fire an event=] named {{addsourcebuffer}} at {{MediaSource/sourceBuffers}}.</li>
-            <li>Return the new object.</li>
+            <li>
+              If |buffer|'s {{SourceBuffer/[[generate timestamps flag]]}} is true, set |buffer|'s {{SourceBuffer/mode}} to {{AppendMode/"sequence"}}.
+              Otherwise, set |buffer|'s {{SourceBuffer/mode}} to {{AppendMode/"segments"}}.
+            </li>
+            <li>[=List/Append=] |buffer| to |this|'s {{MediaSource/sourceBuffers}}.</li>
+            <li>[=Queue a task=] to [=fire an event=] named {{addsourcebuffer}} at |this|'s {{MediaSource/sourceBuffers}}.</li>
```

Comment:
See the comment about this type of prose below, let's make this correct while we're here.

Reply:
@padenot, the events queuing is broken throughout the entire spec... but let's def fix as a follow up.